### ISSUE TITLE
”QIBusPlatformInputContext: invalid portal bus.” and wrong screen 

### ISSUE THREAD

**Issue by twoexem** [flameshot-org/flameshot#2715]
### Flameshot Version

Flameshot v12.0.0 (b4737f8)
Compiled with Qt 5.15.5
linux: 5.18.6-arch1-1
org.kde.Platform: 5.15-21.08

### Installation Type

Flatpak from Github

### Operating System type and version

Arch Linux rolling

### Description

When trying to take a screenshot, Flameshot selects the wrong screen at the wrong position, cropping away half the screen. This only happens with the `gui` option though. Selection works normal. When trying to take a full screen capture, it just spits out the error ”QIBusPlatformInputContext: invalid portal bus.” and doesn't do anything. This only happens under Wayland, it works fine under X11. I've attached a video capture of an attempted screenshot. This happens with the flatpak package (locally and via flathub) and when building from source.

### Steps to reproduce

1. Use a vertical monitor along with KDE Plasma on Wayland
2. Run `flameshot gui` (or really any flameshot option)

### Screenshots or screen recordings

https://user-images.githubusercontent.com/78163379/175387015-40baa3e5-5f52-478e-b662-966dcf7deb8f.mp4

### System Information

Arch Linux rolling
Monitor configuration:
![Screenshot_20220623_220226](https://user-images.githubusercontent.com/78163379/175387221-e6b7256c-7372-4f83-8725-182ee1226f55.png)
KDE Plasma with KWin
Wayland

**Comment by mmahmoudian** [comment]
> neither do shortcuts.

Do you mean keybindings to trigger a screenshot, or do you mean the keybindings to select different tools in Flameshot while taking a screenshot?

**Comment by twoexem** [comment]
I meant the keybinds while taking a capture.
On another note, there's a delay of a few seconds delay between starting a capture (by pressing e.g. the taskbar icon) and the actual GUI appearing. Overall, Flameshot's inconsistent behaviour has rather confused me. Compiling from source locally doesn't work, the AppImage does work, though at first copying to clipboard or saving didn't work either, but that works now, too. In addition to that, occasionaly Flameshot just doesn't launch and says it can't capture the screen. This always happened while playing a game via Proton / Steam. I'm not a programmer myself, so I don't know what in heaven could cause all of that.

**Comment by twoexem** [comment]
Small update: It seems like Flameshot doesn't detect keyboard input *at all* under Wayland. Should I opena new issue for this?

**Comment by mmahmoudian** [comment]
@twoexem 
> Small update: It seems like Flameshot doesn't detect keyboard input *at all* under Wayland. Should I open a new issue for this?

Yes I think that is a separate issue and should be tracked separately

**Comment by twoexem** [comment]
Alright, I will. I'll see if the issue persists with other sources like flatpak, compiled locally etc, then open a new bug report.

**Comment by twoexem** [comment]
I've opened #2731
